private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts

```diff
@@ -376,43 +376,76 @@ export class TemplateServiceClient {
     let existingTemplate: ITemplate = response.result[0];
     let templateName: string = name ? name : existingTemplate.name;
     let existingTags = existingTemplate.tags;
+    
     if (tag) {
       existingTags!.push(tag);
     }
     let tags: string[] | undefined = tag ? existingTags : tagList;
-    let templateState: TemplateState | undefined = isPublished ? TemplateState.live : state;
+    let templateState: TemplateState | undefined =  state;
 
+    if(!version){
```

Comment:
clean coding suggestions - only make checks if-checks against bool variables. names do help with following the code, and 'version' doesn't sound like a flag